### PR TITLE
click: init at 0.3.1

### PR DESCRIPTION
**Pull request by mbode**
###### Motivation for this change
Add [click](https://github.com/databricks/click).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/networking/cluster/click/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    sha256 = "1bl6j1gcl86sgnycfzcyxd1cimnsmgqqsnxw60003182i92zdih5";
+    inherit rev;
+  };
+
+  cargoSha256 = "03vgbkv9xsnx44vivbbhjgxv9drp0yjnimgy6hwm32x74r00k3hj";
+
+  buildInputs = stdenv.lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
+
+  meta = with stdenv.lib; {
+    description = "The \"Command Line Interactive Controller for Kubernetes\"";
```
Comment (nlewo) [inline pkgs/applications/networking/cluster/click/default.nix]: (non blocking) you could use `"""` instead of escaping `"`

**Review by nlewo — CHANGES_REQUESTED** [review]

**Comment by nlewo** [comment]
@mbode we generally don't put the commit id in the package name and the manual don't talk about this.
It would be really nice to ask for an upsteam release... otherwise, do you really need to add the commit id in the name?

File: pkgs/applications/networking/cluster/click/default.nix (hunk 1 above)
Comment (bjornfor) [reply pkgs/applications/networking/cluster/click/default.nix]: I'm pretty sure that should be '' (two single quotes). (The triple quotes thing is Python.)

Comment (nlewo) [reply pkgs/applications/networking/cluster/click/default.nix]: yes, exactly, sorry:)

**Comment by mbode** [comment]
@nlewo I have removed the commit id. The next version is going to be released properly according to https://github.com/databricks/click/issues/34.

**Comment by nlewo** [comment]
@mbode ok,thnaks!
@GrahamcOfBorg build click

**Comment by nlewo** [comment]
@mbode could you either fix the build for `aarch64` or restrict it to `x86` and `darwin`?

**Pull request merged**